### PR TITLE
CI: provide regression testing for stdlib

### PR DESCRIPTION
**Pull request by Undin**
Note, I've created a new batch in regression tests workflow because analysis is quite slow currently for `core_arch` module in `std`.
But #6526 should significantly improve the performance (from 17 to 3 minutes on my machine) of stdlib test, so it's probably worth to merge both batches into a single one after #6526

changelog: provide regression testing for stdlib packages on CI

### CODE REVIEW

**Comment by Undin** [comment]
I've rebased onto master and it became significantly faster with changes #6526 (@vlad20012 thanks!).
But since `regression-1` batch takes more time than any `check` even without `stdlib` project, I keep the second regression batch

File: src/test/kotlin/org/rustPerformanceTests/RsRealProjectTestBase.kt [hunk 1]
```diff
@@ -92,13 +92,19 @@ abstract class RsRealProjectTestBase : RsWithToolchainTestBase() {
...
-fun VirtualFile.findDescendants(filter: (VirtualFile) -> Boolean): ArrayList<VirtualFile> {
+fun VirtualFile.findDescendants(exclude: List<String>, filter: (VirtualFile) -> Boolean): ArrayList<VirtualFile> {
     val result = ArrayList<VirtualFile>()
     VfsUtilCore.visitChildrenRecursively(this,
         object : VirtualFileVisitor<ArrayList<VirtualFile>>() {
             override fun visitFile(file: VirtualFile): Boolean {
-                if (!file.isDirectory && filter(file)) result.add(file)
-                return true
+                return if (file.isDirectory) {
+                    file.name !in exclude
```
Comment (vlad20012) [inline src/test/kotlin/org/rustPerformanceTests/RsRealProjectTestBase.kt:101]: Currently `exclude`  matches relative path prefix (e.g. `src/llvm`) but not a directory name on any level. Also, I don't like changing `findDescendants` to provide such kind of filtering. You can just use `filter` to match a prefix

Comment (Undin) [reply src/test/kotlin/org/rustPerformanceTests/RsRealProjectTestBase.kt]: The corresponding change was introduced to filter out `core_arch` module from stdlib. But after #6526 we don't want to do it so I just dropped the change

**Review by vlad20012 — APPROVED** [review]
bors r+

**Pull request merged**